mlflow/pyfunc/__init__.py

```diff
@@ -1429,7 +1442,13 @@ def batch_predict_fn(pdf):
                 return client.invoke(pdf).get_predictions()
 
         elif env_manager == _EnvManager.LOCAL:
-            if should_use_spark_to_broadcast_file:
+            if is_spark_connect_enabled and not in_databricks_runtime:
+                try:
+                    loaded_model = mlflow.pyfunc.load_model(local_model_path)
+                except OSError:
+                    os.makedirs(local_model_path, exist_ok=True)
+                    loaded_model = mlflow.pyfunc.load_model(model_uri, dst_path=local_model_path)
```

Comment:
In this case, can we make all spark tasks in the same spark worker use the same model path ? (so one spark task download it and other spark task can reuse it, we need to add a lock on the directory path)

Reply:
@HyukjinKwon 

Do you know how to coordinate all spark tasks that runs on the same spark worker to use the same temp directory ?

Reply:
We can make this optimization in follow-up PR.

Reply:
I think, on databricks, we can firstly get the global temp root directory ('/local_disk0/tmp` ), then we create a deterministic temp directory for this model:

`/local_disk0/tmp/{encoded_model_uri}`, then because model_uri is deterministic, then all spark tasks running on this spark worker node will get the same temp directory for this model.

Reply:
sounds good